src/python/pants/backend/experimental/python/lockfile.py

```diff
@@ -124,7 +147,16 @@ async def generate_lockfile(
             output_files=(req.dest,),
         ),
     )
-    return PythonLockfile(result.output_digest, req.dest)
+
+    lockfile_contents = next(
+        iter(await Get(DigestContents, Digest, generated_lockfile.output_digest))
+    )
```

Comment:
We thus far have tended to not put `await` inside expressions like this. It's totally valid, but not as idiomatic because we try to keep the await magic limited in scope.

Instead, maybe rewrite to `lockfile_digest_contents` then use `[0]` to index into it where the variable is used.

Reply:
That's what I'd originally done, it felt a bit more verbose than I'd have liked. Happy to change it back, though.